pkg/nodebootstrap/assets/bootstrap.al2.sh

```diff
@@ -1,7 +1,31 @@
-#!/bin/sh -eu
+#!/bin/bash -e
 
-echo "NODE_IP=$(curl --silent http://169.254.169.254/latest/meta-data/local-ipv4)" > /etc/eksctl/kubelet.local.env
-echo "INSTANCE_ID=$(curl --silent http://169.254.169.254/latest/meta-data/instance-id)" >> /etc/eksctl/kubelet.local.env
+function get_max_pods() {
+  MAX_PODS_FILE="/etc/eksctl/max_pods_map.txt"
+  while read instance_type pods; do
+
+    if  [[ "${instance_type}" == "${1}" ]] && [[ "${pods}" =~ ^[0-9]+$ ]] ; then
+      echo ${pods};
+      return
+    fi ;
+
+    done < "${MAX_PODS_FILE}"
+}
+
+NODE_IP=$(curl --silent http://169.254.169.254/latest/meta-data/local-ipv4)
+INSTANCE_ID=$(curl --silent http://169.254.169.254/latest/meta-data/instance-id)
+INSTANCE_TYPE=$(curl --silent http://169.254.169.254/latest/meta-data/instance-type)
+
+source /etc/eksctl/kubelet.env
+
+MAX_PODS=${MAX_PODS:-$(get_max_pods ${INSTANCE_TYPE})}
+
+set -o nounset
```

Comment:
I see, so this needs to be here because of `MAX_PODS` above... let's find a different way to handle this.

Reply:
e.g. if `MAX_PODS` is set - user it, call the function otherwise. The problem with `set -o` in the middle of a script it is that it's too easy to forget about it, so it's best to just do it at the top.

Reply:
I don't know how to check if the variable is set without getting an error for `unbound variable` unless I move it there. Maybe there's a way I don't know about?

Reply:
There is a way to do it with a [`+` expansion](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02), e.g.:
```
$ env MAX_PODS=456 bash -eu -c 'if [ -z "${MAX_PODS+x}" ] ; then export MAX_PODS=123; fi ; echo $MAX_PODS'   
456
$ bash -eu -c 'if [ -z "${MAX_PODS+x}" ] ; then export MAX_PODS=123; fi ; echo $MAX_PODS'             
123
```

Reply:
I think this looks worse and more error prone than having the set where it is now. I'd rather keep it how it is or perhaps remove the option of passing the MAX_PODS argument since we don't use it or need it anyway

Reply:
I don't mind if it is ugly... It's bash, I find that reliable scripts do get ugly.

We do need to pass it in, as if one wants to use e.g. Weave Net, they need bypass this logic and force a fixed value.

Reply:
I will take a look at this in a bit, I think there is better option...